File: SourceAFIS/Extraction/Model/DotRemover.cs
using System;
using System.Collections.Generic;
using System.Text;
using SourceAFIS.General;

namespace SourceAFIS.Extraction.Model
{
    public sealed class DotRemover : ISkeletonFilter
    {
        public void Filter(SkeletonBuilder skeleton)
        {
            List<SkeletonBuilder.Minutia> removed = new List<SkeletonBuilder.Minutia>();
            foreach (SkeletonBuilder.Minutia minutia in skeleton.Minutiae)
                if (minutia.Ridges.Count == 0)
                    removed.Add(minutia);
            foreach (SkeletonBuilder.Minutia minutia in removed)
                skeleton.RemoveMinutia(minutia);
        }
    }
}

File: java/sourceafis/src/sourceafis/extraction/model/DotRemover.java
/**
 * @author Veaceslav Dubenco
 * @since 20.10.2012
 */
package sourceafis.extraction.model;

import java.util.Iterator;

import sourceafis.general.DetailLogger;

/**
 * 
 */
public class DotRemover implements ISkeletonFilter {

	public DetailLogger.Hook Logger = DetailLogger.off;

	@Override
	public void Filter(SkeletonBuilder skeleton) {
		Iterator<SkeletonBuilderMinutia> iter = skeleton.getMinutiae()
				.iterator();
		while (iter.hasNext()) {
			SkeletonBuilderMinutia minutia = iter.next();
			if (minutia.getRidges().size() == 0) {
				iter.remove();
			}
		}
		Logger.log(skeleton);
	}
}
